code401challenges/src/main/java/GraphExercises/Graph.java

```diff
@@ -72,6 +73,29 @@ public int size(){
         return size;
     }
 
+//  BFT implementation
+    public ArrayList<T> BFTraversal(Graph<T> graph, T root){
+        ArrayList<T> visited = new ArrayList<>();
+        Queue<T> queue = new Queue<>();
+
+        if(root == null){
+            return visited;
+        } else {
+            queue.enqueue(root);
+            visited.add(root);
+        }
+
+        while(!queue.isEmpty()){
```

Comment:
I think part of the issue you're having is that you're never dequeue-ing from the queue, so it's never going to be empty